xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1288 @@
+ *   - This function becomes also called on all add-ons if the master process
+ *     becomes changed.
+ *   - Also every process after StreamInitialize on the addon mode becomes asked
+ *     with _..._ProcessNeededSamplesize(...) about required memory size for the
+ *     output of his data, if no other size is required it can return 0.
+ *
+ * @li From now the processing becomes handled for the different steps with
+ * _..._Process(...).
+ *   - Further it becomes asked with _..._GetDelay(...) about his processing
+ *     time as float value in seconds, needed for video and audio alignment.
+ *
+ * @li On the end of the processing if the source becomes stopped the
+ * StreamDestroy(...) function becomes called on all active processing add-ons.
+ *
+ * @note
+ * The StreamCreate(...) can be becomes called for a new stream before the
+ * previous was closed with StreamDestroy(...) ! To have a speed improve.
+ */
+
+namespace kodi { namespace addon { class CInstanceAudioDSP; }}
+
+extern "C" {
+
+  typedef void* ADSPHANDLE;
+
+  typedef unsigned int AE_DSP_STREAM_ID;
+
+  /*!
+   * @brief Audio DSP add-on error codes
+   */
+  typedef enum
+  {
+    AE_DSP_ERROR_NO_ERROR             = 0,      /*!< @brief no error occurred */
+    AE_DSP_ERROR_UNKNOWN              = -1,     /*!< @brief an unknown error occurred */
+    AE_DSP_ERROR_IGNORE_ME            = -2,     /*!< @brief the used input stream can not processed and add-on want to ignore */
+    AE_DSP_ERROR_NOT_IMPLEMENTED      = -3,     /*!< @brief the method that KODI called is not implemented by the add-on */
+    AE_DSP_ERROR_REJECTED             = -4,     /*!< @brief the command was rejected by the DSP */
+    AE_DSP_ERROR_INVALID_PARAMETERS   = -5,     /*!< @brief the parameters of the method that was called are invalid for this operation */
+    AE_DSP_ERROR_INVALID_SAMPLERATE   = -6,     /*!< @brief the processed samplerate is not supported */
+    AE_DSP_ERROR_INVALID_IN_CHANNELS  = -7,     /*!< @brief the processed input channel format is not supported */
+    AE_DSP_ERROR_INVALID_OUT_CHANNELS = -8,     /*!< @brief the processed output channel format is not supported */
+    AE_DSP_ERROR_FAILED               = -9,     /*!< @brief the command failed */
+  } AE_DSP_ERROR;
+
+  /*!
+   * @brief The possible DSP channels (used as pointer inside arrays)
+   */
+  typedef enum
+  {
+    AE_DSP_CH_INVALID = -1,
+    AE_DSP_CH_FL = 0,
+    AE_DSP_CH_FR,
+    AE_DSP_CH_FC,
+    AE_DSP_CH_LFE,
+    AE_DSP_CH_BL,
+    AE_DSP_CH_BR,
+    AE_DSP_CH_FLOC,
+    AE_DSP_CH_FROC,
+    AE_DSP_CH_BC,
+    AE_DSP_CH_SL,
+    AE_DSP_CH_SR,
+    AE_DSP_CH_TFL,
+    AE_DSP_CH_TFR,
+    AE_DSP_CH_TFC,
+    AE_DSP_CH_TC,
+    AE_DSP_CH_TBL,
+    AE_DSP_CH_TBR,
+    AE_DSP_CH_TBC,
+    AE_DSP_CH_BLOC,
+    AE_DSP_CH_BROC,
+
+    AE_DSP_CH_MAX
+  } AE_DSP_CHANNEL;
+
+  /*!
+   * @brief Present channel flags
+   */
+  typedef enum
+  {
+    AE_DSP_PRSNT_CH_UNDEFINED = 0,
+    AE_DSP_PRSNT_CH_FL        = 1<<0,
```

Comment:
I check, not sure if it works to use a enum in enum (maybe the reason why I done with number), if works is it really better.